src/views/CashlinkManage.vue

```diff
@@ -0,0 +1,342 @@
+                return;
+            }
+            // if there was a funding transaction the cashlink is in the static store.
+            this.retrievedCashlink.networkClient = network;
+            network.on(NetworkClient.Events.API_READY,
+                () => this.status = 'Contacting seed nodes...');
+            network.on(NetworkClient.Events.CONSENSUS_SYNCING,
+                () => this.status = 'Syncing consensus...');
+            network.on(NetworkClient.Events.CONSENSUS_ESTABLISHED,
+                () => this.status = 'Sending transaction...');
+            network.on(NetworkClient.Events.TRANSACTION_PENDING,
+                () => this.status = 'Awaiting receipt confirmation...');
+
+            network.relayTransaction({
+                sender: new Nimiq.Address(this.keyguardRequest.sender).toUserFriendlyAddress(),
+                senderPubKey: this.keyguardResult.publicKey,
+                recipient:  new Nimiq.Address(this.keyguardRequest.recipient).toUserFriendlyAddress(),
+                value: Nimiq.Policy.lunasToCoins(this.keyguardRequest.value),
+                fee: Nimiq.Policy.lunasToCoins(this.keyguardRequest.fee),
+                validityStartHeight: this.keyguardRequest.validityStartHeight,
+                signature: this.keyguardResult.signature,
+                extraData: this.keyguardRequest.data,
+            });
+
+            network.on(NetworkClient.Events.TRANSACTION_RELAYED, async (txInfo: any) => {
+                await CashlinkStore.Instance.put(this.retrievedCashlink!);
+                this.isTxSent = true;
+                window.setTimeout(() => this.state = StatusScreen.State.SUCCESS, StatusScreen.SUCCESS_REDIRECT_DELAY);
+            });
+        }
+    }
+
+    private get title(): string {
+        return this.state === StatusScreen.State.LOADING ? 'Creating your Cashlink' : 'Cashlink created.';
+    }
+
+    private get link(): string {
+        return `${window.location.origin}/${RequestType.CASHLINK}/#${this.retrievedCashlink!.render()}`;
+    }
+
+    private close() {
+        this.$rpc.resolve({
+            address: this.retrievedCashlink!.address.toUserFriendlyAddress(),
+            message: this.retrievedCashlink!.message,
+            status: this.retrievedCashlink!.state,
+        } as PublicCashlink);
+    }
+
+    private copy() {
+        Clipboard.copy(this.link);
+
+        this.copied = true;
+        setTimeout(() => this.copied = false, 800);
+    }
+
+    private get shareText(): string {
+        return encodeURIComponent(`${this.sharePrefix} ${this.link}`);
+    }
+
+    private share() {
+        (navigator as any).share({
+            title: 'Nimiq Cashlink',
+            text: this.sharePrefix,
+            url: this.link,
+        }).catch( (error: Error) => console.log('Error sharing', error));
+    }
+
+    private get telegram(): string {
+        return `https://telegram.me/share/msg?url=${location.host}&text=${this.shareText}`;
```

Comment:
Shouldn't the url be the cashlink?

Reply:
I believe the url is the source url, for telegram to show which app opened it. The cashlink url is included I bthe `shareText`, see its creation above.

Reply:
Yes, the url of the Cashlink is in the `shareText`.